### ISSUE TITLE
Issue running prevalence method with isContaminant

### ISSUE THREAD

**Issue by nmshahir** [benjjneb/decontam#14]
I'm trying to run decontam on my data and I've followed the steps up to

```
 sample_data(ps)$is.neg <- sample_data(ps)$Sample_or_Control == "Control"
 contamdf.prev <- isContaminant(ps, method="prevalence", neg="is.neg")
```

After which I receive an error,
```
Error in if (sum(freq > 0) > 1 && sum(neg, na.rm = TRUE) > 0 && sum(neg,  : 
  missing value where TRUE/FALSE needed
```
Help?
N

**Comment by benjjneb** [comment]
It probably means you have `NA` in your `is.neg` variable.

What is the result of: `table(get_variable(ps, "is.neg"), useNA="always")`

**Comment by nmshahir** [comment]
```
FALSE  TRUE  <NA> 
  316    13     0 
```
so I'm somewhat puzzled by that output

**Comment by benjjneb** [comment]
> if(sum(freq>0)>1 && sum(neg,na.rm=TRUE) > 0 && sum(neg,na.rm=TRUE) < sum(!is.na(neg))) {

Yep, that looks like its probably the problem. It's doing condition checking on `freq` when it shouldn't be be as you are only using the prevalence method. Flagging this as a bug, thanks for the report!

As a temporary workaround, I think that if you just give it a freq variable (can just be random numbers) it should pass the condition checking and work as expected.

**Comment by cedwardson4** [comment]
I tried the random 'freq' column workaround and got the same error. 

I think what is causing this is a sample with zero reads (sample_sum=0). 

No error here:

> head(sort(sample_sums(MycoAZ8_ps_for_decontam5)))
     NTC5 AZ8W.BLK2 AZ8W.BLK1 AZ8W.BLK3    C4.W12    C6.W11 
        5       111       125       213     12267     13343 

Error:

> head(sort(sample_sums(MycoAZ8_ps_for_decontam2)))
 10.4.BLK AZ8B.BLK1   8.1.BLK      NTC2 10.14.BLK AZ8B.BLK2 
        0        32        42        51        85        91

No error when I remove "10.4.BLK"

**Comment by benjjneb** [comment]
Ah that might be it, yeah. We need to take another look at our handling of zero-read samples, this came up in #7 as well.

For now workaround would be to remove those samples first.

**Comment by benjjneb** [comment]
Should be fixed by 82f44ff72f916138c472323b4a15b819e8b0c0e4 which checks for and removes zero-count samples.
